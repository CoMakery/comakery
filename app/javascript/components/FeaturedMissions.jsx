import React from 'react'
import PropTypes from 'prop-types'
import FeaturedMission from './FeaturedMission'
import Slider from 'react-slick'

import headerImg from '../src/images/featured/header.png'
import logo from '../src/images/styleguide/icons/Logo-Footer.svg'
import developersImg from '../src/images/featured/developers.png'
import communityManagersImg from '../src/images/featured/community-managers.png'
import chatImg from '../src/images/featured/chat.svg'
import notificationImg from '../src/images/featured/warning.svg'

const confirmNotification = (
  <div className="confirm-message">
    <div className="confirm-message__box">
      <img src={notificationImg} className="confirm-message__icon" />
      Please confirm your email address to continue
    </div>
  </div>
)

export default class FeaturedMissions extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      topMissions: props.topMissions
    }
  }

  render() {
    const { topMissions } = this.state
    const { moreMissions } = this.props
    const moreMissionsCount = moreMissions.length
    const settings = {
      slidesToShow  : Math.min(4, moreMissionsCount),
      slidesToScroll: 1,
      lazyLoad      : true,
      responsive    : [
        {
          breakpoint: 1440,
          settings  : {
            slidesToShow: Math.min(3, moreMissionsCount)
          }
        },
        {
          breakpoint: 800,
          settings  : {
            slidesToShow: Math.min(2, moreMissionsCount)
          }
        },
        {
          breakpoint: 650,
          settings  : {
            slidesToShow: 1
          }
        }
      ]
    }
    return (
      <div className="featured-missions">
        {!this.props.isConfirmed && confirmNotification}
        <div className="featured-missions__header">
          <div className="intercom">
            <img src={chatImg} />
          </div>
          <div className="featured-missions__header_container">
            <img className="featured-missions__header__img" src={headerImg} />
            <div className="featured-missions__header__description">
              <img className="featured-missions__header__logo" src={logo} />
              <div className="featured-missions__header__title">
                Find Your Mission.<br />
                Prepare For Liftoff.
              </div>
              <div className="featured-missions__header__subtitle">Accelerating Blockchain Adoption</div>
            </div>
          </div>
        </div>
        <div className="featured-missions__content">
          <p className="featured-missions__content__title">
              Featured<br />
            <span className="featured-missions__content__title--big">missions</span>
          </p>
          <p className="featured-missions__content__description">CoMakery Hosts Blockchain Missions We Believe In</p>
          {
            topMissions.map((mission, index) =>
              <FeaturedMission
                key={mission.id}
                id={mission.id}
                float={index % 2 === 0 ? 'left' : 'right'}
                name={mission.name}
                imageUrl={mission.imageUrl}
                description={mission.description}
                projects={mission.projects}
                csrfToken={this.props.csrfToken}
              />
            )
          }
        </div>
        <div className="featured-missions__more">
          <div className="featured-missions__more_container">
            {/* <p className="featured-missions__more__title">
              40+<br />
              <span className="featured-missions__more__title--big">missions</span>
            </p> */}
            <p className="featured-missions__more__description">Discover Missions With Cutting Edge Projects</p>
            <Slider className="featured-missions__gallery" {...settings}>
              { moreMissions.map((mission) =>
                <div key={mission.id}>
                  <div className="gallery-content">
                    <div className="gallery-content__image">
                      <img src={mission.imageUrl} />
                    </div>
                    <div className="gallery-content__title">{mission.name}</div>
                    <div className="gallery-content__description">
                      {mission.projectsCount === 0 && 'No projects'}
                      {mission.projectsCount > 0 && `${mission.projectsCount} ${mission.projectsCount === 1 ? 'project' : 'projects'}` }
                    </div>
                  </div>
                </div>
              )}
            </Slider>
          </div>
        </div>
        <div className="featured-missions__footer">
          <div className="featured-missions__footer_container">
            <div className="featured-missions__footer__stat">
              <div className="featured-missions__footer__stat__num">1000+</div>
              <div className="featured-missions__footer__stat__name">Contributors</div>
            </div>
            <img src={communityManagersImg} className="hide-on-mobile" />
            <div className="featured-missions__footer__stat">
              <div className="featured-missions__footer__stat__num">500+</div>
              <div className="featured-missions__footer__stat__name">PROJECTS</div>
            </div>
            <img src={developersImg} className="hide-on-mobile" />
            <div className="featured-missions__footer__stat">
              <div className="featured-missions__footer__stat__num">1,000,000+</div>
              <div className="featured-missions__footer__stat__name">TOKENS AWARDED</div>
            </div>
            <div className="featured-missions__footer__mobile">
              <img src={communityManagersImg} />
              <img src={developersImg} style={{marginLeft: -30}} />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

FeaturedMissions.propTypes = {
  topMissions : PropTypes.array,
  moreMissions: PropTypes.array,
  csrfToken   : PropTypes.string,
  isConfirmed : PropTypes.bool
}

FeaturedMissions.defaultProps = {
  topMissions : [],
  moreMissions: [],
  csrfToken   : '00',
  isConfirmed : false
}
