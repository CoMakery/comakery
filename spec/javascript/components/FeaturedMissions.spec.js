import React from 'react'
import { mount } from 'enzyme'
import FeaturedMissions from 'components/FeaturedMissions'

describe('FeaturedMissions', () => {
  beforeEach(() => {
    fetch.resetMocks()
  })

  it('renders correctly without props', () => {
    const wrapper = mount(<FeaturedMissions />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.featured-missions__header')).toBe(true)
    expect(wrapper.exists('.featured-missions__content')).toBe(true)
    expect(wrapper.exists('.featured-missions__more')).toBe(true)
    expect(wrapper.exists('.featured-missions__footer')).toBe(true)
  })

  it('grays out page for unconfirmed accounts', () => {
    const wrapper = mount(<FeaturedMissions isConfirmed={false} />)

    expect(wrapper.exists('.grayed-page')).toBe(true)
  })

  it('renders correctly with props', () => {
    const topMissions = [
      {
        id         : 1,
        name       : 'Mission 1',
        description: 'Description 1',
        imageUrl   : '/logo.png',
        symbol     : 'BTC',
        projects   : [
          { id: 1, title: 'Project 1', interested: true },
          { id: 2, title: 'Project 2', interested: false }
        ],
      },
      {
        id         : 2,
        name       : 'Mission 2',
        description: 'Description 2',
        imageUrl   : '/logo.png',
        symbol     : 'BTC',
        projects   : [
          { id: 3, title: 'Project 3', interested: true },
          { id: 4, title: 'Project 4', interested: false }
        ],
      }
    ]
    const moreMissions = [
      {
        id           : 3,
        name         : 'Mission 3',
        imageUrl     : '/logo.png',
        symbol       : 'BTC',
        projectsCount: 3
      },
      {
        id           : 4,
        name         : 'Mission 4',
        imageUrl     : '/logo.png',
        symbol       : 'BTC',
        projectsCount: 2
      },
      {
        id           : 5,
        name         : 'Mission 5',
        imageUrl     : '/logo.png',
        symbol       : 'BTC',
        projectsCount: 3
      },
      {
        id           : 6,
        name         : 'Mission 6',
        imageUrl     : '/logo.png',
        symbol       : 'BTC',
        projectsCount: 2
      }
    ]

    const wrapper = mount(<FeaturedMissions topMissions={topMissions} moreMissions={moreMissions} />)

    expect(wrapper.exists('FeaturedMission[name="Mission 1"]')).toBe(true)
    expect(wrapper.exists('FeaturedMission[name="Mission 2"]')).toBe(true)
    expect(wrapper.exists('Slider[className="featured-missions__gallery"]')).toBe(true)
  })
})
