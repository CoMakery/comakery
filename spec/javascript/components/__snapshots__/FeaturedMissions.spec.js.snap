// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedMissions renders correctly without props 1`] = `
<FeaturedMissions
  csrfToken="00"
  isConfirmed={false}
  moreMissions={Array []}
  topMissions={Array []}
>
  <div
    className="featured-missions"
  >
    <div
      className="confirm-message"
    >
      <div
        className="confirm-message__box"
      >
        <img
          className="confirm-message__icon"
          src="test-file-stub"
        />
        Please confirm your email address to continue
      </div>
    </div>
    <div
      className="featured-missions__header"
    >
      <div
        className="intercom"
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="featured-missions__header_container"
      >
        <img
          className="featured-missions__header__img"
          src="test-file-stub"
        />
        <div
          className="featured-missions__header__description"
        >
          <img
            className="featured-missions__header__logo"
            src="test-file-stub"
          />
          <div
            className="featured-missions__header__title"
          >
            Find Your Mission.
            <br />
            Prepare For Liftoff.
          </div>
          <div
            className="featured-missions__header__subtitle"
          >
            Accelerating Blockchain Adoption
          </div>
        </div>
      </div>
    </div>
    <div
      className="featured-missions__content"
    >
      <p
        className="featured-missions__content__title"
      >
        Featured
        <br />
        <span
          className="featured-missions__content__title--big"
        >
          missions
        </span>
      </p>
      <p
        className="featured-missions__content__description"
      >
        CoMakery Hosts Blockchain Missions We Believe In
      </p>
    </div>
    <div
      className="featured-missions__more"
    >
      <div
        className="featured-missions__more_container"
      >
        <p
          className="featured-missions__more__description"
        >
          Discover Missions With Cutting Edge Projects
        </p>
        <Slider
          className="featured-missions__gallery"
          lazyLoad={true}
          responsive={
            Array [
              Object {
                "breakpoint": 1440,
                "settings": Object {
                  "slidesToShow": 0,
                },
              },
              Object {
                "breakpoint": 800,
                "settings": Object {
                  "slidesToShow": 0,
                },
              },
              Object {
                "breakpoint": 650,
                "settings": Object {
                  "slidesToShow": 1,
                },
              },
            ]
          }
          slidesToScroll={1}
          slidesToShow={0}
        >
          <InnerSlider
            accessibility={true}
            adaptiveHeight={false}
            afterChange={null}
            appendDots={[Function]}
            arrows={true}
            autoplay={false}
            autoplaySpeed={3000}
            beforeChange={null}
            centerMode={false}
            centerPadding="50px"
            className="featured-missions__gallery"
            cssEase="ease"
            customPaging={[Function]}
            dots={false}
            dotsClass="slick-dots"
            draggable={true}
            easing="linear"
            edgeFriction={0.35}
            fade={false}
            focusOnSelect={false}
            infinite={true}
            initialSlide={0}
            lazyLoad={true}
            nextArrow={null}
            onEdge={null}
            onInit={null}
            onLazyLoadError={null}
            onReInit={null}
            pauseOnDotsHover={false}
            pauseOnFocus={false}
            pauseOnHover={true}
            prevArrow={null}
            responsive={
              Array [
                Object {
                  "breakpoint": 1440,
                  "settings": Object {
                    "slidesToShow": 0,
                  },
                },
                Object {
                  "breakpoint": 800,
                  "settings": Object {
                    "slidesToShow": 0,
                  },
                },
                Object {
                  "breakpoint": 650,
                  "settings": Object {
                    "slidesToShow": 1,
                  },
                },
              ]
            }
            rows={1}
            rtl={false}
            slide="div"
            slidesPerRow={1}
            slidesToScroll={1}
            slidesToShow={0}
            speed={500}
            swipe={true}
            swipeEvent={null}
            swipeToSlide={false}
            touchMove={true}
            touchThreshold={5}
            unslick={true}
            useCSS={true}
            useTransform={true}
            variableWidth={false}
            vertical={false}
            waitForAnimate={true}
          >
            <div
              className="slick-slider featured-missions__gallery slick-initialized"
            >
              <div
                className="slick-list"
              >
                <Track
                  centerMode={false}
                  centerPadding="50px"
                  cssEase="ease"
                  currentSlide={0}
                  fade={false}
                  focusOnSelect={null}
                  infinite={true}
                  lazyLoad={true}
                  lazyLoadedList={Array []}
                  listHeight={0}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  rtl={false}
                  slideCount={0}
                  slideHeight={0}
                  slideWidth={NaN}
                  slidesToScroll={1}
                  slidesToShow={0}
                  speed={500}
                  trackStyle={
                    Object {
                      "WebkitTransform": "translate3d(0px, 0px, 0px)",
                      "WebkitTransition": "",
                      "msTransform": "translateX(0px)",
                      "opacity": 1,
                      "transform": "translate3d(0px, 0px, 0px)",
                      "transition": "",
                    }
                  }
                  unslick={true}
                  variableWidth={false}
                  vertical={false}
                >
                  <div
                    className="slick-track"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "WebkitTransform": "translate3d(0px, 0px, 0px)",
                        "WebkitTransition": "",
                        "msTransform": "translateX(0px)",
                        "opacity": 1,
                        "transform": "translate3d(0px, 0px, 0px)",
                        "transition": "",
                      }
                    }
                  />
                </Track>
              </div>
            </div>
          </InnerSlider>
        </Slider>
      </div>
    </div>
    <div
      className="featured-missions__footer"
    >
      <div
        className="featured-missions__footer_container"
      >
        <div
          className="featured-missions__footer__stat"
        >
          <div
            className="featured-missions__footer__stat__num"
          >
            1000+
          </div>
          <div
            className="featured-missions__footer__stat__name"
          >
            Contributors
          </div>
        </div>
        <img
          className="hide-on-mobile"
          src="test-file-stub"
        />
        <div
          className="featured-missions__footer__stat"
        >
          <div
            className="featured-missions__footer__stat__num"
          >
            500+
          </div>
          <div
            className="featured-missions__footer__stat__name"
          >
            PROJECTS
          </div>
        </div>
        <img
          className="hide-on-mobile"
          src="test-file-stub"
        />
        <div
          className="featured-missions__footer__stat"
        >
          <div
            className="featured-missions__footer__stat__num"
          >
            1,000,000+
          </div>
          <div
            className="featured-missions__footer__stat__name"
          >
            TOKENS AWARDED
          </div>
        </div>
        <div
          className="featured-missions__footer__mobile"
        >
          <img
            src="test-file-stub"
          />
          <img
            src="test-file-stub"
            style={
              Object {
                "marginLeft": -30,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</FeaturedMissions>
`;
